src/stores/subgraphStore.ts

```diff
@@ -0,0 +1,254 @@
+import { defineStore } from 'pinia'
+import { computed, ref } from 'vue'
+
+import { t } from '@/i18n'
+import { SubgraphNode } from '@/lib/litegraph/src/litegraph'
+import type { NodeError } from '@/schemas/apiSchema'
+import type {
+  ComfyNode,
+  ComfyWorkflowJSON,
+  NodeId
+} from '@/schemas/comfyWorkflowSchema'
+import type {
+  ComfyNodeDef as ComfyNodeDefV1,
+  InputSpec
+} from '@/schemas/nodeDefSchema'
+import { api } from '@/scripts/api'
+import { useDialogService } from '@/services/dialogService'
+import { useWorkflowService } from '@/services/workflowService'
+import { useExecutionStore } from '@/stores/executionStore'
+import { useCanvasStore } from '@/stores/graphStore'
+import { ComfyNodeDefImpl } from '@/stores/nodeDefStore'
+import { UserFile } from '@/stores/userFileStore'
+import {
+  ComfyWorkflow,
+  LoadedComfyWorkflow,
+  useWorkflowStore
+} from '@/stores/workflowStore'
+
+async function confirmOverwrite(name: string): Promise<boolean | null> {
+  return await useDialogService().confirm({
+    title: t('sideToolbar.workflowTab.confirmOverwriteTitle'),
+    type: 'overwrite',
+    message: t('sideToolbar.workflowTab.confirmOverwrite'),
+    itemList: [name]
+  })
+}
+
+export class SubgraphBlueprint extends ComfyWorkflow {
+  static override readonly basePath = 'subgraphs/'
+
+  hasPromptedSave: boolean = false
+
+  validateSubgraph() {
+    if (!this.activeState?.definitions)
+      throw new Error(
+        'The root graph of a subgraph blueprint must consist of only a single subgraph node'
+      )
+    const subgraphs = this.activeState.definitions.subgraphs
+    const nodes = this.activeState.nodes
```

Comment:
If you're feeling fancy...
```suggestion
    const { subgraphs } = this.activeState.definitions
    const { nodes } = this.activeState
```